swingset/src/main/java/com/nqadmin/swingset/utils/SSVersion.java

```diff
@@ -0,0 +1,208 @@
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Contributors:
+ *   Prasanth R. Pasala
+ *   Brian E. Pangburn
+ *   Diego Gil
+ *   Man "Bee" Vo
+ *   Ernie R. Rael
+ * ****************************************************************************/
+package com.nqadmin.swingset.utils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.logging.log4j.Logger;
+
+// TODO: could add isNull() method for 0.0.0-SNAPSHOT
+// TODO: could add getMajor(), ... methods
+// TODO: could handle optional "<branch>branch-name</branch> i the pom
+
+/**
+ * Version string is of the form "{@literal <digits>.<digits>.<digits>-SNAPSHOT}"
+ * where "-SNAPSHOT" is optional and missing components are 0.
+ * <p>
+ *{@linkplain SSVersion} is comparable, for example,
+ * <pre>
+ * {@code
+ * if (SSVersion.get().compareTo(SSVersion.get("4.0.11")) >= 0) {
+ *     // Do this for SwingSet version 4.0.11 and later
+ * }
+ * assert SSVersion.get("4.0").compareTo(SSVersion.get("4.0.0")) == 0;
+ * assert SSVersion.get("4.0.11-SNAPSHOT").compareTo(SSVersion.get("4.0.11")) < 0;
+ * }
+ * </pre>
+ */
+public class SSVersion implements Comparable<SSVersion> {
+	/** Log4j Logger for component */
+	private static final Logger logger = SSUtils.getLogger();
+
+	/** Name of SwingSet version resource file. */
+	private final static String SS_VERSION_FILENAME = "swingsetVersion.properties";
+
+	/** Name of version key in resource file. */
+	private final static String VERSION_KEY = "version";
+
+	/** Use all zeros version if there's a parsing error. */
+	private final static String ERROR_VERSION = "0.0.0-SNAPSHOT";
+
+	/** Versions that end with this are considered snapshots. */
+	private final static String SNAP = "-SNAPSHOT";
+
+	private static SSVersion SINGLETON;
+
+	/**
+	 * Returns the singleton SSVersion of the running SwingSet library.
+	 * If the version property is not found, or there is a parse error,
+	 * then 0.0.0-SNAPSHOT is used.
+	 * @return version
+	 */
+	public static SSVersion get() {
+		if (SINGLETON == null) {
+			SINGLETON = parse(getVersionProperty());
+		}
+		return SINGLETON;
+	}
+
+	/**
+	 * Returns the SSVersion resulting from parsing the {@linkplain sVer}
+	 * argument. If there is a parse error, then 0.0.0-SNAPSHOT is used.
+	 * The return values are useful for comparing.
+	 * @param sVer version string to parse
+	 * @return version
+	 */
+	public static SSVersion get(String sVer) {
+		return parse(sVer);
+	}
+
+	/** Version number components. */
+	private final List<Integer> versionSequence;
+	/** Is the version a snapshot? */
+	private final boolean isSnapshot;
+
+	private SSVersion(List<Integer> _versionSequence, boolean _isSnapshot) {
```

Comment:
Know it's not required, but I try to add JavaDoc for protected/private methods just for the developer benefit.